File: src/main/java/io/github/shirohoo/library/application/persistance/user/UserLoanHistoryJpaRepository.java
package io.github.shirohoo.library.application.persistance.user;

import io.github.shirohoo.library.application.domain.user.UserLoanHistory;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryJpaRepository extends JpaRepository<UserLoanHistory, Long> {
    boolean existsByBookTitleAndIsReturn(String bookTitle, boolean isReturn);
}

File: src/main/kotlin/io/github/shirohoo/library/application/persistence/user/UserLoanHistoryJpaRepository.kt
package io.github.shirohoo.library.application.persistence.user

import io.github.shirohoo.library.domain.user.UserLoanHistory
import io.github.shirohoo.library.domain.user.UserLoanStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserLoanHistoryJpaRepository : JpaRepository<UserLoanHistory, Long> {
    fun existsByBookTitleAndStatus(bookTitle: String, status: UserLoanStatus): Boolean
    fun findAllByStatus(status: UserLoanStatus): List<UserLoanHistory>
}
